### PR TITLE
make validation core functions private

### PR DESCRIPTION
**Pull request by mgovers**
Idem to #787 but now for validation core

relates to #784

### CODE REVIEW

File: src/power_grid_model/validation/rules.py [hunk 1]
```diff
@@ -70,12 +70,12 @@
...
 )
 from power_grid_model.validation.utils import (
-    eval_expression,
-    get_indexer,
-    get_mask,
-    get_valid_ids,
-    nan_type,
-    set_default_value,
+    _eval_expression,
+    _get_indexer,
```
Comment (figueroa1395) [inline src/power_grid_model/validation/rules.py:74]: Isn't this one intended to be public?

Comment (figueroa1395) [inline src/power_grid_model/validation/rules.py:74]: Isn't this one intended to be public?

File: src/power_grid_model/validation/rules.py [hunk 2]
```diff
@@ -70,12 +70,12 @@
...
+    _get_mask,
+    _get_valid_ids,
+    _nan_type,
```
Comment (figueroa1395) [inline src/power_grid_model/validation/rules.py:77]: Would this be useful for the users to be public? Or is there a python side function that does something similar already?

**Review by figueroa1395 — APPROVED** [review]
Just double checking a couple.

File: src/power_grid_model/validation/rules.py (hunk 1 above)
Comment (mgovers) [reply src/power_grid_model/validation/rules.py:74]: we want people to use `PowerGridModel.get_indexer()` instead (https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.PowerGridModel.get_indexer). 

In addition, the `power_grid_model.validation.utils` module where it is declared has a module-level docstring that says
```
Utilities used for validation. Only errors_to_string() is intended for end users.
```

This file actually exposes these utilities even though it only wishes to use it internally

File: src/power_grid_model/validation/rules.py (hunk 2 above)
Comment (mgovers) [reply src/power_grid_model/validation/rules.py:77]: we have https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.typing.ComponentMetaData.nans

File: src/power_grid_model/validation/rules.py (hunk 1 above)
Comment (figueroa1395) [reply src/power_grid_model/validation/rules.py:74]: Oops I confused this one with the other `get_indexer`. Great find.

Comment (nitbharambe) [reply src/power_grid_model/validation/rules.py:74]: @zhen0427 You can implement this in sensitivity directly instead of importing.

Comment (figueroa1395) [reply src/power_grid_model/validation/rules.py:74]: @nitbharambe @zhen0427 but `power_grid_model.get_indexer` is still public. Or is it different to the other indexer in the validation functions?

**Pull request merged**